### PR TITLE
Native canvas image

### PR DESCRIPTION
**Pull request by CedricGuillemet**
Work continues on Native Canvas with DOM image being abstracted for native.
And fixed an issue with 0 sized image for fullscreen GUI

### CODE REVIEW

File: gui/src/2D/controls/image.ts [hunk 1]
```diff
@@ -312,7 +312,7 @@ export class Image extends Control {
         const width = this._domImage.width;
         const height = this._domImage.height;
 
-        // Shoudl abstract platform instead of using LastCreatedEngine
+        // Should abstract platform instead of using LastCreatedEngine
         const engine = Engine.LastCreatedEngine;
```
Comment (sebavan) [inline gui/src/2D/controls/image.ts:316]: I wonder if it should be along those lines: 

this._host?.getScene()?.getEngine() || Engine.LastCreatedEngine; ?

File: gui/src/2D/controls/image.ts [hunk 2]
```diff
@@ -511,7 +511,12 @@ export class Image extends Control {
             value = this._svgCheck(value);
         }
 
-        this._domImage = document.createElement("img");
+        // Should abstract platform instead of using LastCreatedEngine
+        const engine = Engine.LastCreatedEngine;
```
Comment (sebavan) [inline gui/src/2D/controls/image.ts:515]: Same I wonder if it should be along those lines:

this._host?.getScene()?.getEngine() || Engine.LastCreatedEngine; ?

File: src/Engines/thinEngine.ts [hunk 3]
```diff
@@ -649,6 +649,15 @@ export class ThinEngine {
         return ThinEngine._createCanvas(width, height);
     }
 
+    /**
+     * Create an image to use with canvas
+     * @return IImage interface
+     */
+     public createCanvasImage(): IImage
```
Comment (sebavan) [inline src/Engines/thinEngine.ts:656]: Nit picking but unsure about the name, let s add @bghgary as well as he is good with naming which is weird for a dev :-)

**Review by sebavan — APPROVED** [review]
Only couple nits and comments

**Review by bghgary — COMMENTED** [review]
Sorry, didn't finish the review yesterday, but I have some minor comments.

File: gui/src/2D/controls/image.ts [hunk 4]
```diff
@@ -312,8 +312,8 @@ export class Image extends Control {
         const width = this._domImage.width;
         const height = this._domImage.height;
 
-        // Shoudl abstract platform instead of using LastCreatedEngine
-        const engine = Engine.LastCreatedEngine;
+        // Should abstract platform instead of using LastCreatedEngine
+        const engine = this._host?.getScene()?.getEngine() || Engine.LastCreatedEngine;
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply gui/src/2D/controls/image.ts:316]: I agree. I'll fix that in a future PR.

File: src/Engines/nativeEngine.ts [hunk 5]
```diff
@@ -1841,6 +1841,10 @@ export class NativeEngine extends Engine {
     }
 
     public createDynamicTexture(width: number, height: number, generateMipMaps: boolean, samplingMode: number): InternalTexture {
+        // it's not possible to create 0x0 texture sized. Many bgfx methods assume texture size is at least 1x1(best case).
+        // Worst case is getting a crash/assert.
+        width = Math.max(width, 1);
+        height = Math.max(height, 1);
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply src/Engines/nativeEngine.ts:1847]: Yes, it happens :) https://github.com/BabylonJS/Babylon.js/blob/2213db320d66875322159a328106acb7f3025da5/gui/src/2D/advancedDynamicTexture.ts#L1021

[the remark this answers is not in the document]
Comment (bghgary) [reply src/Engines/nativeEngine.ts:1847]: If someone gets the width/height after creating a 0x0, it will be the wrong value?

**Pull request merged**